src/vision/status_quo/Alan_tries_to_debug_a_hang.md

```diff
@@ -0,0 +1,134 @@
+# 😱 Status quo stories: Alan tries to debug a hang
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Alan's startup has officially launched and YouBuy is live for the world to use.
+The whole team is very excited especially as this will be their first use of Rust in production! 
+Normally, as a .Net shop, they would have written the entire application in C# but because of the scalability and latency requirements on their inventory service, they decided to write that microservice in Rust utilizing the `async` features they've heard so much about. 
+
+The day's excitement soon turns into concern as reports begin coming into support of customers who can't checkout.
+After a few cases, a pattern begins to emerge: when a customer is trying to buy the last available item, the checkout process hangs forever.
+
+Alan suspects there is an issue with the lock used in the inventory service to prevent multiple people from buying the last available item at the same time.
+With this hunch, he builds the latest code and opens this local dev environment to conduct some tests.
+Soon enough, Alan has a repro of the bug.
+
+With the broken environment still running, he decides to use a debugger to see if he can confirm his theory.
+In the past, Alan has used Visual Studio's debugger to [diagnose a very similar issue in a C# application he wrote](https://devblogs.microsoft.com/visualstudio/how-do-i-debug-async-code-in-visual-studio/#is-there-a-way-to-better-visualize-tasks-and-async-code-flow).
+The debugger was able to show him all the async Tasks currently waiting, their call stacks and what resource they were waiting on.
+
+Alan hasn't used a debugger with Rust before, usually a combination of the strict compiler and a bit of manual testing has been enough to fix all the bugs he's previously encountered.
+He does a quick Google search to see what debugger he should use and decides to go with `gdb` since that's already installed on his system and it sounds like it should work.
```

Comment:
```suggestion
He does a quick Google search to see what debugger he should use and decides to go with `gdb` because it is already installed on his system and sounds like it should work.
```